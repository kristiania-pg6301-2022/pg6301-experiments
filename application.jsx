export function Application() {
    return <h1>Hello React</h1>;
}
