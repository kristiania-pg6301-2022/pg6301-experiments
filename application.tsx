import * as React from "react";

export function Application({someone}: {someone: string}) {
    return <h1>Hello {someone}</h1>;
}
