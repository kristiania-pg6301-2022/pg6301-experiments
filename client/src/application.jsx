import * as React from "react";
import {BrowserRouter, Link, Route, Routes} from "react-router-dom";
import {Books} from "./books";
import {useTitle} from "./useTitle";

function ApplicationHeader() {
    return <header><h1><Link to={"/"}>Kristiania Library</Link></h1></header>;
}

function ApplicationMenu() {
    return <nav>
        <ul>
            <li><Link to={"books"}>View books</Link></li>
            <li>View authors</li>
            <li><Link to={"books/create"}>Add book</Link></li>
        </ul>
    </nav>;
}

function FrontPage() {
    useTitle("Kristiania Library", true);
    return <h1>Welcome</h1>;
}

function ApplicationMain() {
    return <main>
        <div>
            <Routes>
                <Route path={"/"} element={<FrontPage/>}/>
                <Route path={"books/*"} element={<Books/>}/>
                <Route path={"*"} element={<h1>Not found</h1>}/>
            </Routes>
        </div>
    </main>;
}

export function Application() {
    return <BrowserRouter>
        <main>
            <ApplicationHeader/>
            <ApplicationMenu/>
            <ApplicationMain/>
        </main>
    </BrowserRouter>
}

