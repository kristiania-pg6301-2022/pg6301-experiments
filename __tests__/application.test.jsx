import React from "react";
import ReactDOM from "react-dom";
import {Application} from "../application";


describe("application", () => {
    it("renders", () => {
        const element = document.createElement("div");
        ReactDOM.render(<Application />, element);
        expect(element.innerHTML).toMatchSnapshot();
    })
})
