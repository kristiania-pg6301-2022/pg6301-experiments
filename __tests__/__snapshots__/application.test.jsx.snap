// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application renders 1`] = `"<h1>Hello React</h1>"`;
